specreduce/tests/test_extinction.py

```diff
@@ -56,14 +59,16 @@ def test_missing_extinction_unit():
     """
     Test creation of custom model from Quantity arrays
     """
-    wave = np.linspace(0.3, 2.0, 50)
-    extinction = 1. / wave
-    ext = AtmosphericExtinction(extinction=extinction, spectral_axis=wave * u.um)
-    assert(len(ext.extinction_mag) > 0)
-    assert(len(ext.transmission) > 0)
+    with pytest.warns(AstropyUserWarning):
```

Comment:
Ideally, you only want the context manager covering the line that actually emits the warning.

Reply:
i have made the edits to do this. only the lines emitting the warnings are now within the context handlers.